lib/AST/ASTImporter.cpp

```diff
@@ -3286,6 +3286,38 @@ ExpectedDecl ASTNodeImporter::VisitIndirectFieldDecl(IndirectFieldDecl *D) {
   return ToIndirectField;
 }
 
+static std::tuple<unsigned int, unsigned int>
+getFriendCountAndPosition(FriendDecl *FD) {
+  unsigned int FriendCount = 0;
+  llvm::Optional<unsigned int> FriendPosition;
+  auto *RD = cast<CXXRecordDecl>(FD->getLexicalDeclContext());
+  if (FD->getFriendType()) {
+    QualType TypeOfFriend = FD->getFriendType()->getType().getCanonicalType();
+    for (FriendDecl *FoundFriend : RD->friends()) {
+      if (FoundFriend == FD) {
+        FriendPosition = FriendCount;
+        ++FriendCount;
+      } else if (FoundFriend->getFriendType() &&
```

Comment:
What if the `FoundFriend` is a decl (i.e. it has a `getFriendDecl`)?

Reply:
The same ("equivalent") friends are counted, if it is decl it is not equivalent.